ci/deploy/npm.py

```diff
@@ -0,0 +1,164 @@
+            )
+        node_version = f"{crate_version}.{build_identifier}"
+        is_development = True
+    return Version(rust=rust_version, node=node_version, is_development=is_development)
+
+
+def get_crate_version() -> str:
+    with open("src/environmentd/Cargo.toml", "r") as envd:
+        for line in envd:
+            if line.startswith("version ="):
+                return line.split("=").pop().strip().replace('"', "")
+    raise ValueError("version not found")
+
+
+def build_package(version: Version, project_path: Path) -> Path:
+    spawn.runv(["bin/wasm-build", str(project_path)])
+    package_path = project_path / "pkg"
+    shutil.copyfile(str(ROOT_PATH / "LICENSE"), str(package_path / "LICENSE"))
+    with open(package_path / "package.json", "r+") as package_file:
+        package = json.load(package_file)
+        # Since all packages are scoped to the MaterializeInc org, names don't need prefixes
+        package["name"] = package["name"].replace("/mz-", "/")
+        package["version"] = version.node
+        package["license"] = "SEE LICENSE IN 'LICENSE'"
+        package["repository"] = "github:MaterializeInc/materialize"
+        package_file.seek(0)
+        json.dump(package, package_file, indent=2)
+    return package_path
+
+
+def release_package(version: Version, package_path: Path) -> None:
+    with open(package_path / "package.json", "r") as package_file:
+        package = json.load(package_file)
+    name = package["name"]
+    dist_tag = "dev" if version.is_development else "latest"
+    if version_exists_in_npm(name, version):
+        logger.warning("%s %s already released, skipping.", name, version.node)
+        return
+    else:
+        logger.info("Releasing %s %s", name, version.node)
+        set_npm_credentials(package_path)
+        spawn.runv(
+            ["npm", "publish", "--access", "public", "--tag", dist_tag],
+            cwd=package_path,
+        )
+
+
+def build_all(version: Version, *, do_release: bool = True) -> None:
+    for path_str in PROJECT_PATHS:
+        project_path = Path(path_str)
+        logger.info("Building %s @ %s", project_path, version.node)
+        package_path = build_package(version, project_path)
+        logger.info("Built %s", project_path)
+        if do_release:
+            release_package(version, package_path)
+            logger.info("Released %s", package_path)
+        else:
+            logger.info("Skipping release for %s", package_path)
+
+
+def version_exists_in_npm(name: str, version: Version) -> bool:
+    quoted = urllib.parse.quote(name)
+    res = requests.get(f"https://registry.npmjs.org/{quoted}/{version.node}")
+    if res.status_code == 404:
+        # This is a new package
+        return False
+    res.raise_for_status()
+    return True
+
+
+def set_npm_credentials(package_path: Path) -> None:
+    with open(package_path / ".npmrc", "w") as npmrc:
+        npmrc.write("//registry.npmjs.org/:_authToken=${NPM_TOKEN}\n")
```

Comment:
```suggestion
    (package_path / ".npmrc").write_text("//registry.npmjs.org/:_authToken=${NPM_TOKEN}\n")
```

Reply:
Man, I didn't realize that `Path` offered this API. Thanks!

Reply:
You bet! `Path` is my favorite Python type.